state/project/Project.kt

```diff
@@ -33,11 +34,11 @@ import kotlinx.coroutines.delay
 import mu.KotlinLogging
 
 @OptIn(ExperimentalTime::class)
-class Project internal constructor(val path: Path, private val projectMgr: ProjectManager) : Navigable<PathState> {
+class Project internal constructor(val path: Path, private val projectMgr: ProjectManager, preferenceMgr: PreferenceManager) : Navigable<PathState> {
```

Comment:
We do quite a lot of work to bring `PreferenceManager` into scope where we need it: this will likely be the case for a lot of preferences in the future as we generally need it at the point that we make a decision deep within Studio's logic. Does it make more sense for us to pass it through everywhere we need it, or to access it via the `StudioState` object?

Reply:
Glad you asked!

Intuitively, the preferences are part of the global state of the program, and therefore I think we should access it via the `StudioState` object.

There is precedent for this - check out the way(s) that we access the currently active `Theme`.

Another hint that leads in this direction: once we perform the refactoring that allows Studio to maintain concurrent active windows/ project tabs, `ProjectManager` will no longer be truly "global" (or, at least, will no longer be a singleton), while `PreferenceManager` will continue to be - it will be a singleton that is shared by all windows/tabs, and it won't be possible to open two Preference dialogs concurrently.

Reply:
That makes sense! I was wondering how we would handle preferences across multiple Studio tabs, and having a single global state of preferences is both easier to reason about for users and is in line with other IDEs. Done accordingly.